tools/flang2/flang2exe/ll_write.cpp

```diff
@@ -738,60 +738,40 @@ ll_write_local_objects(FILE *out, LL_Function *function)
-                object->type->str, name);
-        fprintf(out, "\t%s.ptr = alloca i32*, align 4\n", name);
+        fprintf(out, "\t%s.ptr = alloca ptr, align 4\n", name);
         fprintf(out, "\t%s.count = alloca i32, align 4\n", name);
-        fprintf(out, "\tstore i32 %d, i32* %s.count, align 4\n",
+        fprintf(out, "\tstore i32 %d, ptr %s.count, align 4\n",
                 (int)(ll_type_bytes(object->type) / 4), name);
-        fprintf(out, "\t%s.temp0 = bitcast i32* %s.temp to i8*\n", name, name);
-        fprintf(out, "\t%s.temp1 = bitcast i32** %s.ptr to i8**\n", name, name);
-        fprintf(out, "\tstore i8* %s.temp0, i8** %s.temp1, align 4\n", name,
+        fprintf(out, "\tstore ptr %s, ptr %s.ptr, align 4\n", name,
                 name);
         fprintf(out, "\tbr label %%L.st.init.%04d.1\n", curr_nan_label_count);
         fprintf(out, "L.st.init.%04d.1:\n", curr_nan_label_count);
-        fprintf(out, "\t%s.temp2 = load i32, i32* %s.count, align 4\n", name,
+        fprintf(out, "\t%s.temp = load i32, ptr %s.count, align 4\n", name,
                 name);
-        fprintf(out, "\t%s.temp3 = icmp sle i32 %s.temp2, 0\n", name, name);
+        fprintf(out, "\t%s.temp0 = icmp sle i32 %s.temp, 0\n", name, name);
         fprintf(out,
-                "\tbr i1 %s.temp3, label %%L.st.init.%04d.0,"
+                "\tbr i1 %s.temp0, label %%L.st.init.%04d.0,"
                 " label %%L.st.init.%04d.2\n",
                 name, curr_nan_label_count + 1, curr_nan_label_count);
         fprintf(out, "L.st.init.%04d.2:\n", curr_nan_label_count);
-        fprintf(out, "\t%s.temp4 = load i32*, i32** %s.ptr, align 4\n", name,
+        fprintf(out, "\t%s.temp1 = load ptr, ptr %s.ptr, align 4\n", name,
                 name);
-        fprintf(out, "\tstore i32 %s, i32* %s.temp4, align 4\n",
+        fprintf(out, "\tstore i32 %s, ptr %s.temp1, align 4\n",
                 POWER_STACK_32_BIT_NAN, name);
-        fprintf(out, "\t%s.temp5 = bitcast i32* %s.temp4 to i8*\n", name, name);
-        fprintf(out, "\t%s.temp6 = getelementptr i8, i8* %s.temp5, i32 4\n",
+        fprintf(out, "\t%s.temp2 = getelementptr i8, ptr %s.temp1, i32 4\n",
                 name, name);
-        fprintf(out, "\t%s.temp7 = bitcast i32** %s.ptr to i8**\n", name, name);
-        fprintf(out, "\tstore i8* %s.temp6, i8** %s.temp7, align 4\n", name,
+        fprintf(out, "\tstore ptr %s.temp2, ptr %s.ptr, align 4\n", name,
                 name);
-        fprintf(out, "\t%s.temp8 = load i32, i32* %s.count, align 4\n", name,
+        fprintf(out, "\t%s.temp3 = load i32, ptr %s.count, align 4\n", name,
                 name);
-        fprintf(out, "\t%s.temp9 = sub i32 %s.temp8, 1\n", name, name);
-        fprintf(out, "\tstore i32 %s.temp9, i32* %s.count, align 4\n", name,
+        fprintf(out, "\t%s.temp4 = sub i32 %s.temp3, 1\n", name, name);
+        fprintf(out, "\tstore i32 %s.temp4, ptr %s.count, align 4\n", name,
```

Comment:
Could you add a `llvm_ir_correct` test case that demonstrates this LLVM IR pattern?

Reply:
This code is only enabled in the PowerPC target.

Reply:
> Could you add a `llvm_ir_correct` test case that demonstrates this LLVM IR pattern?

This code is only enabled in the PowerPC target.

Reply:
OK thanks.